app/components/post/post.js

```diff
@@ -486,175 +243,82 @@ class Post extends PureComponent {
+            isLastReply,
             post,
             renderReplies,
-            theme
+            theme,
+            user
         } = this.props;
         const style = getStyleSheet(theme);
-        const PROFILE_PICTURE_SIZE = 32;
+        const isPostSystemMessage = isSystemMessage(post);
+        const pendingOrFailed = isPostPendingOrFailed(post);
+        const selected = this.state && this.state.selected ? style.selected : null;
+        const showReplyBar = !!commentedOnPost; //eslint-disable-line no-implicit-coercion
 
-        let profilePicture;
-        let displayName;
-        let messageStyle;
-        if (isSystemMessage(post)) {
-            profilePicture = (
-                <View style={style.profilePicture}>
-                    <MattermostIcon
-                        color={theme.centerChannelColor}
-                        height={PROFILE_PICTURE_SIZE}
-                        width={PROFILE_PICTURE_SIZE}
+        return (
+            <View style={[style.container, this.props.style, selected]}>
+                <View style={[style.profilePictureContainer, (pendingOrFailed && style.pendingPost)]}>
+                    <PostProfilePicture
+                        enablePostIconOverride={config.EnablePostIconOverride === 'true'}
+                        fromWebHook={post.props && post.props.from_webhook === 'true'}
+                        isSystemMessage={isPostSystemMessage}
+                        onViewUserProfile={this.viewUserProfile}
+                        overrideIconUrl={post.props && post.props.override_icon_url}
+                        theme={theme}
+                        user={user}
                     />
                 </View>
-            );
-
-            displayName = (
-                <FormattedText
-                    id='post_info.system'
-                    defaultMessage='System'
-                    style={style.displayName}
-                />
-            );
-
-            messageStyle = [style.message, style.systemMessage];
-        } else if (post.props && post.props.from_webhook) {
-            if (config.EnablePostIconOverride === 'true') {
-                const icon = post.props.override_icon_url ? {uri: post.props.override_icon_url} : webhookIcon;
-                profilePicture = (
-                    <View style={style.profilePicture}>
-                        <Image
-                            source={icon}
-                            style={{
-                                height: PROFILE_PICTURE_SIZE,
-                                width: PROFILE_PICTURE_SIZE,
-                                borderRadius: PROFILE_PICTURE_SIZE / 2
-                            }}
+                <View style={style.messageContainerWithReplyBar}>
+                    {!commentedOnPost && this.renderReplyBar(style)}
+                    <View style={[style.rightColumn, (commentedOnPost && isLastReply && style.rightColumnPadding)]}>
+                        <PostHeader
```

Comment:
Can either of these components listen to the store for these props or implement the event handlers themselves? We're passing over 30 props between the two of them